src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/ProblemDetailsResultFilter.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc.Infrastructure
+{
+    public sealed class ProblemDetailsResultFilter : IAlwaysRunResultFilter, IOrderedFilter
+    {
+        private readonly IDictionary<int, Func<ActionContext, ProblemDetails>> _problemDetailsFactory;
+        private readonly ILogger<ProblemDetailsResultFilter> _logger;
+
+        /// <summary>
+        /// Gets the filter order. Defaults to -1000.
+        /// </summary>
+        public int Order => -1000;
```

Comment:
It's worth documenting here that we intend for this to run **Early**. Result filters tend to care about what `TYPE` of result is being executed. So since we're changing the type, we want to do it early.